src/expansion_context.mli

```diff
@@ -5,6 +5,17 @@ module Base : sig
   (** Return the code path for the given context *)
   val code_path : t -> Code_path.t
 
+  (** Return the input name for the given context.
+  For a context [ctxt], the input name given by [input_name ctxt] tends to coincide with
```

Comment:
Maybe this should start by explicitly stating how `input_name` is determined, e.g. (if I'm not mistaken) the file name of the driver's argument.

It might actually be worth moving the relevant parts to the `code_path` documentation right above.

How about making it more like a list of properties for both, i.e. in short, something like:
```ocaml
(** ...
  In Driver, Deriving and Extension the context is initialized so that the [file_path] component of
  the [code_path] is determined from the first location found in the input AST. That means that:
  - It's the empty string in empty structure or signature
  - It can be altered by line directives
  etc...
  *)
val code_path : ...

(** ...
  In Driver, Deriving and Extension, the context argument is initialized so that the [input_name]
  matches the input filename passed to the driver on the command line.
  - It is ["_none_"] when rewriting ...
  - It is not affected by line directives
  - It may have a value even for empty files
  etc...
  *)
val input_name : ...
```

Reply:
That's a very nice idea, thanks! I've structured it that way now.